vortex-vector/src/null/vector_mut.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+//! Definition and implementation of [`NullVectorMut`].
+
+use vortex_dtype::{DType, Nullability};
+
+use super::NullVector;
+use crate::VectorMutOps;
+
+/// A mutable vector of null values.
+///
+/// Since a "null" value does not require any data storage, the nulls are stored internally with a
+/// single `length` counter.
+///
+/// The immutable equivalent of this type is [`NullVector`].
+#[derive(Debug, Clone, Copy)]
+pub struct NullVectorMut {
+    pub(super) len: usize,
+}
+
+impl NullVectorMut {
+    /// Creates a new mutable vector of nulls with the given length.
+    pub fn new(len: usize) -> Self {
+        Self { len }
+    }
+}
+
+impl VectorMutOps for NullVectorMut {
+    type Immutable = NullVector;
+
+    fn nullability(&self) -> Nullability {
+        Nullability::Nullable
+    }
+
+    fn dtype(&self) -> DType {
+        DType::Null
+    }
+
+    fn len(&self) -> usize {
+        self.len
+    }
+
+    fn capacity(&self) -> usize {
+        usize::MAX
+    }
+
+    fn reserve(&mut self, _additional: usize) {
+        // We do not allocate memory for `NullVector`, so this is a no-op.
+    }
+
+    fn extend_from_vector(&mut self, other: &Self::Immutable) {
+        self.len += other.len;
+    }
+
+    fn freeze(self) -> Self::Immutable {
+        NullVector::new(self.len)
+    }
+
+    fn split_off(&mut self, at: usize) -> Self {
+        assert!(
+            at <= self.capacity(),
+            "split_off out of bounds: {:?} <= {:?}",
+            at,
+            self.capacity(),
+        );
+
+        // TODO(connor): This is wrong (https://docs.rs/bytes/latest/src/bytes/bytes_mut.rs.html#320-335)
```

Comment:
Either add a `todo!()` that fails at runtime, or fix the bug. I don't think we should merge code that will silently do the wrong thing